packages/@aws-cdk/core/lib/token.ts

```diff
@@ -146,6 +159,19 @@ export class Tokenization {
     return TokenMap.instance().lookupList(l);
   }
 
+  /**
+   * Reverse any value into a Resolvable, if possible
+   *
+   * In case of a string, the string must not be a concatenation.
+   */
+  public static reverse(x: any): IResolvable | undefined {
```

Comment:
I am not 100% sure this has tons of value as a public API. Can we make it internal for now?

Reply:
Not really. If we want to use it in other modules, it needs to be public (Adam's needed this when implementing a guard that reverses Tokens in CodeBuild, but ended up reimplementing himself).

Plus, it seems only fair that if the API to encode a token to a literal is public, the API to reverse that encoding should also be public.